docs/sync.js

```diff
@@ -139,15 +140,17 @@ class User {
 	static login(server, username, password, callback) {}
 
 	/**
-	 * Login a sync user using an external login provider.
+	 * Register/login a sync user using an external login provider.
 	 * @param {string} server - authentication server
-	 * @param {string} provider - The provider type
-	 * @param {string} providerToken - The access token for the given provider
+     * @param {object} options - options, containing the following:
+	 * @param {string} options.provider - The provider type
+	 * @param {string} options.providerToken - The access token for the given provider
+     * @param {object} [options.userInfo] - A map containing additional data required by the provider
```

Comment:
Probably to handle both the old API exposed by the JS client, and the new one introduced by this change. Also, because the userInfo is completely optional, and unused by most authentication delegates.